scripts/publish-plugin.sh

```diff
@@ -27,25 +27,35 @@ fi
 # Tar up the plugin
 tar -czf ${PLUGIN_PACKAGE_PATH} -C ${WORK_PATH} .
 
+# Push the current AWS credentials, since we'll need them to assume a second role.
+export INITIAL_AWS_ACCESS_KEY_ID="${AWS_ACCESS_KEY_ID}"
+export INITIAL_AWS_SECRET_ACCESS_KEY="${AWS_SECRET_ACCESS_KEY}"
+
 # rel.pulumi.com is in our production account, so assume that role first
 CREDS_JSON=$(aws sts assume-role \
                  --role-arn "arn:aws:iam::058607598222:role/UploadPulumiReleases" \
                  --role-session-name "upload-plugin-pulumi-resource-vault" \
                  --external-id "upload-pulumi-release")
 
-# Use the credentials we just assumed
+# Use the credentials we just assumed.
 export AWS_ACCESS_KEY_ID=$(echo ${CREDS_JSON}     | jq ".Credentials.AccessKeyId" --raw-output)
 export AWS_SECRET_ACCESS_KEY=$(echo ${CREDS_JSON} | jq ".Credentials.SecretAccessKey" --raw-output)
 export AWS_SECURITY_TOKEN=$(echo ${CREDS_JSON}    | jq ".Credentials.SessionToken" --raw-output)
 
-echo "Uploading ${PLUGIN_PACKAGE_NAME}..."
+echo "Uploading ${PLUGIN_PACKAGE_NAME} to s3://rel.pulumi.com..."
 
 aws s3 cp --only-show-errors "${PLUGIN_PACKAGE_PATH}" "s3://rel.pulumi.com/releases/plugins/${PLUGIN_PACKAGE_NAME}"
 
 # Assume the role to publish plugins to s3://get.pulumi.com. We upload the plugins to two buckets while
 # we transition to only publishing/serving them from get.pulumi.com.
 echo "Uploading ${PLUGIN_PACKAGE_NAME} to s3://get.pulumi.com..."
 
+# Restore the initial AWS credentials we had, since the assumed role doesn't have the
+# ability to assume this other role to publish into a different bucket.
+export AWS_ACCESS_KEY_ID="${INITIAL_AWS_ACCESS_KEY_ID}"
+export AWS_SECRET_ACCESS_KEY="${INITIAL_AWS_SECRET_ACCESS_KEY}"
+unset AWS_SECURITY_TOKEN
```

Comment:
Is it worth unsetting all old `AWS_*` envvars first, before setting the secondary creds?
Preferably in one unset command using a glob pattern? e.g. https://stackoverflow.com/a/52128244/2274669

Reply:
That's a good idea. I couldn't get the snippet from SO to work on my machine, probably because I'm using `zsh`. But it's probably for the best, since if we were removing `AWS_*` environment variables it could easily lead to problems we don't want. (e.g. blowing away `AWS_PROFILE` or `AWS_CONFIG_FILE`.)

So nuking the security key and token env vars is probably safer in that regard.
https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html

Reply:
Changes LGTM. You should unset first before exporting since as it currently reads, it will cancel out the secondary creds set right above it.